### PR TITLE
Add MPU Ad for amp and canonical

### LINKED ISSUE #6608: Ads: Support MPUs for Canonical Ads and Render an MPU underneath Top Stories for AMP and Canonical
_This task is dependent upon the `mpu` slot type is implemented by GNL. Ensure that this has taken place before carrying out this work._

We already support MPUs for AMP #6512 

**Is your feature request related to a problem? Please describe.**
We currently support leaderboard ads in Canonical and we also need to support MPU ads. We need to render MPU ads on both Canonical and AMP pages.

There is also an AMP version of this task #6512 but most of the implementation for Canonical will be handled by the GNL ads script which we use https://github.com/bbc/gnl-ngas

**Describe the solution you'd like**
_Canonical_

- Combine `containers/Ad/Canonical/AdSlot.jsx` into `containers/Ad/Canonical/index.jsx`
- Allow `uniqueId` to be passed into `CanonicalAd`
- `uniqueId` should now allow one of 2 values `leaderboard` or `mpu` so update PropType to be an enum.
- gnl-ngas should handle this change when the values are passed to `registerSlot`. 
- The correct amount of space should be reserved on the page as per UX designs https://app.zeplin.io/project/5c8a5e021d454124ef90669f/screen/5e790b9088869d6a1a17279b - currently we reseve the correct amount of space in terms of height and width for a leaderboard ad - we need to make sure that if we are rendering an MPU that we reserve the correct amount of space for an MPU instead. @tochwill update this with values.

_AMP and Canonical_
- We should then render an MPU below the top stories section, for both AMP and Canonical pages.

_n.b. there will be another issue to work with UX to render this MPU in the correct place within story promos._

**Testing notes**
- Should match the designs https://app.zeplin.io/project/5c8a5e021d454124ef90669f/screen/5e790b9088869d6a1a17279b

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by thekp**
Resolves #6608

**Overall change:**
_Add a mpu ad to front page under top stories_

**Code changes:**

- _Consoldiated `Canonical/AdSlot.jsx` into `Canonical/index.jsx`_
- _Conditionally render mpu ad under top stories on front page_
- _Updated unit tests so that they use a service that is intended to display ads (mundo, afrique)_
- _Added unit tests for mpu ads_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by thekp** [comment]
URL: http://localhost:7080/mundo?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
Note: Rendering MPU and leaderboard on Canonical

image

**Comment by thekp** [comment]
URL: http://localhost:7080/mundo.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
Note: Rendering leaderboard ad but not mpu ad on AMP (could be because of failed response to fetch mpu ad)



image

File: src/app/containers/Ad/Canonical/index.jsx [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 import { C_LUNAR_LIGHT } from '@bbc/psammead-styles/colours';
 import { GEL_GROUP_4_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';
 import { Helmet } from 'react-helmet';
-import AdSlot from './AdSlot';
 
 const LEADERBOARD_HEIGHT = '5.5rem';
 const LEADERBOARD_HEIGHT_GROUP_4_5 = '9rem';
+const MPU_HEIGHT = '9rem';
```
Comment (thekp) [inline src/app/containers/Ad/Canonical/index.jsx]: This height is used to set a min-height for the MPU placeholder/grey-container surrounding the ad. 

I'm currently not sure what to set this value to be.

File: src/app/containers/Ad/index.test.jsx [hunk 2]
```diff
@@ -36,79 +36,79 @@ describe('Ad Container', () => {
 
   describe('Snapshots', () => {
     shouldMatchSnapshot(
-      'should correctly render an AMP ad',
-      <ServiceContextProvider service="pidgin">
+      'should correctly render an AMP leaderbaord ad',
```
Comment (DenisHdz) [inline src/app/containers/Ad/index.test.jsx]:
```suggestion
      'should correctly render an AMP leaderboard ad',
```

File: src/app/containers/Ad/index.test.jsx [hunk 3]
```diff
@@ -36,79 +36,79 @@ describe('Ad Container', () => {
...
+            <AdContainer slotType="leaderboard" />
           </ToggleContext.Provider>
         </RequestContextProvider>
       </ServiceContextProvider>,
     );
 
     shouldMatchSnapshot(
-      'should not render an AMP ad for News',
-      <ServiceContextProvider service="news">
+      'should correctly render a Canonical leaderbaord ad',
```
Comment (DenisHdz) [inline src/app/containers/Ad/index.test.jsx]:
```suggestion
      'should correctly render a Canonical leaderboard ad',
```

File: src/app/containers/Ad/Canonical/index.jsx (hunk 1 above)
Comment (DenisHdz) [reply src/app/containers/Ad/Canonical/index.jsx]: @tochwill Do you know these values?

File: src/app/containers/Ad/Canonical/index.test.jsx [hunk 4]
```diff
@@ -23,5 +23,12 @@ describe('CanonicalAds Ads', () => {
         <CanonicalAd slotType="leaderboard" />
       </BrowserRouter>,
     );
+
+    shouldMatchSnapshot(
+      'should correctly render an Canonical mpu ad with dotcom-bootstrap script',
+      <BrowserRouter>
+        <CanonicalAd slotType="leaderboard" />
```
Comment (DenisHdz) [inline src/app/containers/Ad/Canonical/index.test.jsx]:
```suggestion
      'should correctly render a Canonical mpu ad with dotcom-bootstrap script',
      <BrowserRouter>
        <CanonicalAd slotType="mpu" />
```

**Review by DenisHdz — COMMENTED** [review]
Looking good so far, just some typos

File: src/app/containers/Ad/Canonical/index.jsx (hunk 1 above)
Comment (itimirichard) [reply src/app/containers/Ad/Canonical/index.jsx]: I've changed this to use the value of the shorter MPU ad for now (which is 15.625rem). We can update this when get a final answer from MPU support! @DenisHdz

File: src/app/containers/Ad/Canonical/index.jsx [hunk 5]
```diff
@@ -1,35 +1,63 @@
...
 import { Helmet } from 'react-helmet';
-import AdSlot from './AdSlot';
 
 const LEADERBOARD_HEIGHT = '5.5rem';
 const LEADERBOARD_HEIGHT_GROUP_4_5 = '9rem';
+const MPU_HEIGHT = '15.625rem';
 
-const StyledAdDiv = styled.div`
-  background-color: ${C_LUNAR_LIGHT};
+const leaderboardStlyes = css`
```
Comment (tochwill) [inline src/app/containers/Ad/Canonical/index.jsx]:
```suggestion
const leaderboardStyles = css`
```

File: src/app/containers/Ad/Canonical/index.jsx [hunk 6]
```diff
@@ -1,35 +1,63 @@
...
-    {/* Loading dotcom-bootstrap.js here instead of CanonicalAdBootstrapJs to avoid it loading on live */}
-    {/* This can be moved once we allow the script to load on live */}
-    <StyledAdDiv>
+const mpuStyles = css`
+  min-height: ${MPU_HEIGHT};
+`;
+
+const AdContainer = styled.div`
+  background-color: ${C_LUNAR_LIGHT};
+  ${({ slotType }) => (slotType === 'mpu' ? mpuStyles : leaderboardStlyes)}
```
Comment (tochwill) [inline src/app/containers/Ad/Canonical/index.jsx]:
```suggestion
  ${({ slotType }) => (slotType === 'mpu' ? mpuStyles : leaderboardStyles)}
```

File: src/app/containers/Ad/Amp/index.test.jsx [hunk 7]
```diff
@@ -67,27 +72,58 @@ describe('AMP Ads', () => {
...
         expect(ad).toHaveAttribute('data-a4a-upgrade-type');
-        expect(ad).toHaveAttribute('json', JSON.stringify(adJsonAttributes));
+        expect(ad).toHaveAttribute(
+          'json',
+          JSON.stringify(adJsonAttributes('leaderboard')),
+        );
+      });
+    });
+
+    it('should render two mpu ads', () => {
```
Comment (tochwill) [inline src/app/containers/Ad/Amp/index.test.jsx]:
```suggestion
    it('should render one mpu ad', () => {
```

**Review by tochwill — COMMENTED** [review]
Looks good! I think let's get this merged in and then consider the height/width of the background in a follow up PR.

File: src/app/containers/Ad/Canonical/index.jsx (hunk 6 above)
Comment (thekp) [reply src/app/containers/Ad/Canonical/index.jsx]: My spelling 😭

**Review by DenisHdz — APPROVED** [review]

**Review by tochwill — APPROVED** [review]

**Comment by greenc05** [comment]
Could we get this rendered in a html section element with a label please? Even though this will be hidden to AT we should still output semantically correct HTML... Or will ths come as part of https://github.com/bbc/simorgh/pull/7025#discussion_r450157934

**Comment by DenisHdz** [comment]
> Could we get this rendered in a html section element with a label please? Even though this will be hidden to AT we should still output semantically correct HTML... Or will ths come as part of [#7025 (comment)](https://github.com/bbc/simorgh/pull/7025#discussion_r450157934)

This will be done in @OlgaLyubin's PR

**Comment by PriyaKR** [comment]
The MPU ads are not displayed on amp pages http://localhost:7080/pidgin.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int

**Comment by tochwill** [comment]
> The MPU ads are not displayed on amp pages http://localhost:7080/pidgin.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int

Hey Priya, there's an issue in progress to fix this. Could you toggle set the ads toggle to 'true' locally at https://github.com/bbc/simorgh/blob/latest/src/app/lib/config/toggles/index.js#L4 and try again?

**Comment by PriyaKR** [comment]
> > The MPU ads are not displayed on amp pages http://localhost:7080/pidgin.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
> 
> Hey Priya, there's an issue in progress to fix this. Could you toggle set the ads toggle to 'true' locally at https://github.com/bbc/simorgh/blob/latest/src/app/lib/config/toggles/index.js#L4 and try again?

@tochwill yeah with the toggles value set to true i can only see the leaderboard ads and not the amp ads.
MPU_ads
MPU_ads

**Comment by thekp** [comment]
I can see `mpu` slotType working on latest so I'll move my PR back in progress and looking into resolving the issue Priya highlighted in this PR ⛑️ 🚒 

image

**Comment by thekp** [comment]
Regarding MPU Ads not showing on AMP. It is because the campaign is no longer valid so the AdsOps team are going to disable this ad. We are going to continue with merging

**Comment by thekp** [comment]
I can see MPU Ad on AMP now, GNL has resolved the issue on their side
url: http://localhost:7080/pidgin.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
and toggle ads on `.../config/toggles`
image

**Pull request merged**